service/internal/checksum/middleware_compute_input_checksum_test.go

```diff
@@ -487,24 +437,9 @@ func TestComputeInputPayloadChecksum(t *testing.T) {
 				expectErr:      "failed to parse algorithm",
 				expectBuildErr: true,
 			},
-			"no algorithm require checksum unseekable stream": {
-				optionsFn: func(o *computeInputPayloadChecksum) {
-					o.RequireChecksum = true
-				},
-				buildInput: middleware.BuildInput{
-					Request: func() *smithyhttp.Request {
-						r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
-						r.URL, _ = url.Parse("http://example.aws")
-						r = requestMust(r.SetStream(bytes.NewBuffer([]byte("hello world"))))
-						return r
-					}(),
-				},
-				expectErr:      "unseekable stream is not supported",
-				expectBuildErr: true,
-			},
 			"http unseekable stream": {
 				initContext: func(ctx context.Context) context.Context {
-					return internalcontext.SetChecksumInputAlgorithm(ctx, string(AlgorithmCRC32))
+					return internalcontext.SetChecksumInputAlgorithm(ctx, string(AlgorithmSHA1))
```

Comment:
I approve of the changes to typos you make on the PR. I don't like the random changes to algorithms which have no functional impact. They increase the diff without adding value to the PR, since the functionality (get an algorithm and use it) has already been tested and is not being changed.